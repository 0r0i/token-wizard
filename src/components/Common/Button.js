import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
