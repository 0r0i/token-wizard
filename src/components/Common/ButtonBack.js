import React from 'react'

export const ButtonBack = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonBack ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    Back
  </button>
)
