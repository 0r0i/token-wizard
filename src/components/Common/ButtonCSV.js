import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
