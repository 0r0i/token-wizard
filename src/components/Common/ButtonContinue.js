import React from 'react'

export const ButtonContinue = ({
  buttonText = 'Continue',
  disabled = false,
  extraClassName = '',
  onClick,
  type = 'button'
}) => (
  <button className={`sw-ButtonContinue ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
