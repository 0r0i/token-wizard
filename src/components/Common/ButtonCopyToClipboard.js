import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
