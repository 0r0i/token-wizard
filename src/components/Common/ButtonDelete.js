import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
