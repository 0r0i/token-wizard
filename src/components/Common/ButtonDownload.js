import React from 'react'

export const ButtonDownload = ({ disabled, onClick, extraClass = '' }) => (
  <button className={`sw-ButtonDownload ${extraClass}`} disabled={disabled} onClick={onClick} type="button">
    Download
  </button>
)
