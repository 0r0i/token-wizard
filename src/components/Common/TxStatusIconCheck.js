import React from 'react'

export const TxStatusIconCheck = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="12" height="10" viewBox="0 0 12 10">
    <path fill="#000" fillRule="evenodd" d="M4.3 6.6L1.7 4 .3 5.4l4 4L11.8 2 10.4.6z" />
  </svg>
)
