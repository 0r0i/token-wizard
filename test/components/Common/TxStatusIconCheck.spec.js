import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { TxStatusIconCheck } from '../../../src/components/Common/TxStatusIconCheck'

configure({ adapter: new Adapter() })

describe(`TxStatusIconCheck`, () => {
  it(`should be a <svg> tag`, () => {
    const wrapper = shallow(<TxStatusIconCheck />)

    expect(wrapper.type()).toBe('svg')
  })

  it(`should render TxStatusIconCheck component`, () => {
    // Given
    const component = render.create(<TxStatusIconCheck />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
