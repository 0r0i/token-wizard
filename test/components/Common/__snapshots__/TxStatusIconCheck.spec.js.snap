// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxStatusIconCheck should render TxStatusIconCheck component 1`] = `
<svg
  height="10"
  viewBox="0 0 12 10"
  width="12"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M4.3 6.6L1.7 4 .3 5.4l4 4L11.8 2 10.4.6z"
    fill="#000"
    fillRule="evenodd"
  />
</svg>
`;
